sdks/java/io/influxdb/src/main/java/org/apache/beam/sdk/io/influxdb/InfluxDBIO.java

```diff
@@ -0,0 +1,624 @@
+        }
+
+        public Write withDatabase(String database) {
+            return builder().setDatabase(database).build();
+        }
+
+        public Write withSslEnabled(boolean sslEnabled){
+            return builder().setSslEnabled(sslEnabled).build();
+        }
+        public Write withSslInvalidHostNameAllowed(Boolean value) {
+            return builder().setSslInvalidHostNameAllowed(value).build();
+        }
+        public Write withNoOfElementsToBatch(Integer noOfElementsToBatch) {
+            return builder().setNoOfElementsToBatch(noOfElementsToBatch).build();
+        }
+        public Write withFlushDuration(Integer flushDuration) {
+            return builder().setFlushDuration(flushDuration).build();
+        }
+        public Write withRetentionPolicy(String rp) {
+            return builder().setRetentionPolicy(rp).build();
+        }
+
+
+
+        private class InfluxWriterFn<T> extends DoFn<T, Void> {
+
+            private final Write spec;
+            private InfluxDB connection;
+
+            InfluxWriterFn(Write write) {
+                this.spec = write;
+            }
+
+            @Setup
+            public void setup() throws Exception {
+                connection = getConnection(spec.dataSourceConfiguration(), sslInvalidHostNameAllowed(), sslEnabled());
+                int flushDuration = spec.flushDuration() != null  ? spec.flushDuration(): DEFAULT_FLUSH_DURATION;
+                int noOfBatchPoints = spec.noOfElementsToBatch() != null ? spec.noOfElementsToBatch() : DEFAULT_NUMBER_OF_DURATION;
+                connection.enableBatch(BatchOptions.DEFAULTS.actions(noOfBatchPoints).flushDuration(flushDuration));
+                connection.setDatabase(spec.database());
+            }
+
+            @StartBundle
+            public void startBundle(StartBundleContext c) {
+            }
+
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+//                    Point point = Point.measurementByPOJO(c.element().getClass()).addFieldsFromPOJO(c.element()).build();
+                    connection.write(c.element().toString());
+            }
+
+            @FinishBundle
+            public void finishBundle() throws Exception {
+                connection.flush();
+            }
+
+            @Teardown
+            public void tearDown() throws Exception {
+                if (connection != null) {
+                    connection.flush();
+                    connection.close();
+                    connection = null;
+                }
+            }
+
+            @Override
+            public void populateDisplayData(DisplayData.Builder builder) {
+                builder.delegate(Write.this);
+            }
+
+            private final Integer DEFAULT_NUMBER_OF_DURATION = 1000;
+            private final Integer DEFAULT_FLUSH_DURATION = 100;
+
+        }
+    }
+
+    public static OkHttpClient.Builder getUnsafeOkHttpClient() {
```

Comment:
@kennknowles Sorry, do you know what normal Beam policy is around dealing with certs is?